google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta2/st/STBigQueryStorageLongRunningWriteTest.java

```diff
@@ -0,0 +1,170 @@
+import com.google.cloud.bigquery.storage.v1beta2.JsonStreamWriter;
+import com.google.cloud.bigquery.storage.v1beta2.TableName;
+import com.google.cloud.bigquery.storage.v1beta2.it.ITBigQueryStorageLongRunningTest;
+import com.google.cloud.bigquery.testing.RemoteBigQueryHelper;
+import com.google.protobuf.Descriptors;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+import org.threeten.bp.LocalDateTime;
+
+public class STBigQueryStorageLongRunningWriteTest {
+  public enum Complexity {
+    SIMPLE,
+    COMPLEX
+  }
+
+  private static final Logger LOG =
+      Logger.getLogger(ITBigQueryStorageLongRunningTest.class.getName());
+  private static final String LONG_TESTS_ENABLED_PROPERTY =
+      "bigquery.storage.enable_long_running_tests";
+
+  private static BigQueryWriteClient client;
+  private static String parentProjectId;
+  private static final String DATASET = RemoteBigQueryHelper.generateDatasetName();
+  private static final String TABLE = "testtable";
+  private static final String DESCRIPTION = "BigQuery Write Java long test dataset";
+
+  private static TableInfo tableInfo;
+  private static String tableId;
+  private static BigQuery bigquery;
+
+  private static JSONObject MakeJsonObject(Complexity complexity) throws IOException {
+    JSONObject object = new JSONObject();
+    // size: (1, simple)(2,complex)()
+    switch (complexity) {
+      case SIMPLE:
+        object.put("test_str", "aaa");
+        object.put("test_numerics", new JSONArray(new String[] {"1234", "-900000"}));
+        object.put("test_datetime", String.valueOf(LocalDateTime.now()));
+        break;
+      case COMPLEX:
+        // TODO(jstocklass): Make a better json object that doesn't break the format rules.
+        break;
+      default:
+        break;
+    }
+    return object;
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (client != null) {
+      client.close();
+    }
+    if (bigquery != null) {
+      RemoteBigQueryHelper.forceDelete(bigquery, DATASET);
+      LOG.info("Deleted test dataset: " + DATASET);
+    }
+  }
+
+  @Test
+  public void testDefaultStream()
+      throws IOException, InterruptedException, ExecutionException,
+          Descriptors.DescriptorValidationException {
+    // TODO(jstocklass): Set up a default stream. Write to it for a long time,
+    // (a few minutes for now) and make sure that everything goes well, report stats.
+    parentProjectId = String.format("projects/%s", ServiceOptions.getDefaultProjectId());
+    RemoteBigQueryHelper bigqueryHelper = RemoteBigQueryHelper.create();
+    bigquery = bigqueryHelper.getOptions().getService();
+    client = BigQueryWriteClient.create();
+    DatasetInfo datasetInfo =
```

Comment:
You need to create the dataset with a unique id to the run, so that if the test is run at the same time, they won't step onto each other, and it is easier to check results.

Reply:
I've changed the DATASET string to be generated in the test run so it should be good now. The generateDatasetName function should work for that right?

Reply:
SG.